main.go

```diff
@@ -241,114 +46,112 @@ func main() {
 			os.Args[0], flagUsage, version)
 		flag.PrintDefaults()
 	}
-	flag.Parse()
+}
 
+func setupLogger() {
+	logrus.SetOutput(os.Stderr)
+	logrus.SetLevel(logrus.DebugLevel)
+	logrus.SetFormatter(&logrus.TextFormatter{
+		FullTimestamp: true,
+	})
 	if !useVerboseLogging() {
 		logrus.SetLevel(logrus.InfoLevel)
 	}
-
 	if jsonLog {
 		logrus.SetFormatter(&logrus.JSONFormatter{})
 	}
+}
 
-	switch {
-	case testMode && flag.NArg() == 1:
-	case flag.NArg() == 5:
-	default:
+func useVerboseLogging() bool {
+	return verboseOutput || (len(os.Getenv(envNameVerbose)) > 0)
+}
+
+func main() {
+	var err error
+	ctx := context.Background()
+	ctx, stop := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
+	defer stop()
+
+	flag.Parse()
+	if flag.NArg() == 0 {
 		flag.Usage()
 		os.Exit(2)
 	}
 
-	configFile := flag.Arg(0)
+	setupLogger()
 
+	WaitForKeepalivedTermination(ctx, stop)
```

Comment:
I'm not 100% sure that this needs to be in `main()`, but it probably doesn't hurt.

Reply:
Yeah not entirely happy with this either, but I don't really want to pass the cancel function further down the stack